### PR TITLE
ASoC: SOF: cleanup for upstream

### PR DESCRIPTION
**Pull request by plbossart**
Add comments and remove topology code that's not currently needed.

The trace will not be renamed though, it's too invasive at this
point. I'll change the commit messages

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

File: sound/soc/sof/topology.c [hunk 1]
```diff
@@ -2432,7 +2410,7 @@ static struct snd_soc_tplg_ops sof_tplg_ops = {
 
 	/* DAI link - used for any driver specific init */
 	.link_load	= sof_link_load,
-	.link_unload	= sof_link_unload,
+	/* .link_unload	is not currently used */
```
Comment (ranj063) [inline sound/soc/sof/topology.c]: @plbossart I think @RanderWang 's latest patch uses link_unload to unset the link dma stream

File: sound/soc/sof/topology.c [hunk 2]
```diff
@@ -2340,13 +2322,6 @@ static int sof_route_load(struct snd_soc_component *scomp, int index,
 	return ret;
 }
 
-static int sof_route_unload(struct snd_soc_component *scomp,
```
Comment (ranj063) [inline sound/soc/sof/topology.c:2504]: @plbossart also can we please leave this on for now. I need this for my free_topology() fix changes. I will submit the PR by tomorrow.

Comment (plbossart) [reply sound/soc/sof/topology.c:2504]: ok. Advance notice that I will freeze the code for upstream on Friday 12pm Central time. Whatever is not merged by then will be considered as an update for upstream. Tick tock

Comment (ranj063) [reply sound/soc/sof/topology.c:2504]: @plbossart this change is good. we can remove it.

File: sound/soc/sof/topology.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/sof/topology.c]: yes, removed this change

File: sound/soc/sof/topology.c (hunk 2 above)
Comment (plbossart) [reply sound/soc/sof/topology.c:2504]: done

**Pull request merged**